wagtail/wagtailadmin/static_src/wagtailadmin/js/page-editor.js

```diff
@@ -391,6 +391,28 @@ $(function() {
         insertRichTextDeleteControl(this);
     });
 
+    /* Make user confirm before leaving the editor if there are unsaved changes */
+    var initialData = $('#page-edit-form').serialize();
+    function dataHasChanged() {
+        return $('#page-edit-form').serialize() != initialData;
+    }
+
+    window.addEventListener('beforeunload', function(event) {
+        // Ignore if the user clicked one of the submit buttons
+        var trigger = $(event.originalTarget.activeElement);
```

Comment:
Hello, @kaedroho!

Do you use Firefox? [`event.originalTarget` is mozilla-specific](https://developer.mozilla.org/ru/docs/Web/API/Event/originalTarget). I think you should use [`event.target`](https://developer.mozilla.org/en-US/docs/Web/API/Event/target) instaead.